### PR TITLE
oomstore(json-schema): add snapshot dependence and unit test

### PR DESCRIPTION
**Pull request by lianxmfor**

### CODE REVIEW

File: pkg/oomstore/types/apply/validate_test.go [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+			description: "invaid kind: spelling error",
+			entity: &apply.Entity{
+				Kind:        "Entit",
+				Name:        "user",
+				Description: "~",
+			},
+			wantError: fmt.Errorf(`Entity user: /kind: "Entit" must equal "Entity"`),
+		},
+		{
+			description: "invalid kind: empty kind ",
```
Comment (YikSanChan) [inline pkg/oomstore/types/apply/validate_test.go:30]: remove tailing space

**Review by YikSanChan — COMMENTED** [review]
```go
func (e *Entity) Validate() error
func (g *Group) Validate() error
func (f *Feature) Validate() error
```

The implementations are almost identical, can we have just one method to represent them 3? Such as

```go
func Validate(data interface{}, schema jsonschema.Schema) error
```

**Comment by lianxmfor** [comment]
> ```go
> func (e *Entity) Validate() error
> func (g *Group) Validate() error
> func (f *Feature) Validate() error
> ```
> 
> The implementations are almost identical, can we have just one method to represent them 3? Such as
> 
> ```go
> func Validate(data interface{}, schema jsonschema.Schema) error
> ```

It might work, I'll try it:
```go
func Validate(errPrefix string, data interface{}, schema *jsonschema.Schema) error
```

**Comment by lianxmfor** [comment]
> > ```go
> > func (e *Entity) Validate() error
> > func (g *Group) Validate() error
> > func (f *Feature) Validate() error
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > The implementations are almost identical, can we have just one method to represent them 3? Such as
> > ```go
> > func Validate(data interface{}, schema jsonschema.Schema) error
> > ```
> 
> It might work, I'll try it:
> 
> ```go
> func Validate(errPrefix string, data interface{}, schema *jsonschema.Schema) error
> ```

done

File: pkg/oomstore/types/apply/validate_test.go (hunk 1 above)
Comment (lianxmfor) [reply pkg/oomstore/types/apply/validate_test.go]: good catch! fixed

**Review by YikSanChan — APPROVED** [review]
lgtm

**Pull request merged**